.github/workflows/build-and-test.yml

```diff
@@ -5,19 +5,94 @@ on:
     branches: ['master']
 
 jobs:
-  build-and-test:
+  build:
+    name: Build
     runs-on: ubuntu-latest
-
     steps:
       - uses: oven-sh/setup-bun@v1
         with:
           bun-version: 1.0.0
       - uses: actions/checkout@v3
+      - name: Restore cached dependencies
+        id: dependency-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: node_modules/
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('bun.lockb') }}
       - name: Install dependencies
-        run: bun i
+        run: bun i --frozen-lockfile
+      - name: Save dependency cache
```

Comment:
The downsides of bun have begun... can't use the cypress github action. If stuff like this keeps coming up we are going to have to switch back to node/npm.

Reply:
yeah.... 😬

Reply:
🙃